Orange/widgets/visualize/owmosaic.py

```diff
@@ -476,31 +456,21 @@ def reset_graph(self):
         self.update_graph()
 
     def set_color_data(self):
-        if self.data is None or len(self.data) < 2 or len(self.data.domain.attributes) < 1:
+        if self.data is None or len(self.data) < 2:
```

Comment:
This doesn't look like a necessary condition for setting the color, but I suspect it was added as a fix for some crash that occurs sometime later if colors are set -- instead of fixing it at its proper place. I would leave this as it this.

Reply:
No, this indeed is a bug and should be fixed (gonna open an issue). Not the removed condition, but the existing one. `len(self.data) < 2` is used only here which causes wrong value of `self.discrete_data` when resetting the graph. Domains (current `self.data.domain` and the domain `self.discrete_data` was built from) then differ and the widget crashes.